pkg/ebpf/net_events.go

```diff
@@ -230,40 +241,50 @@ func (t *Tracee) processNetEvents(ctx gocontext.Context) {
 					continue
 				}
 
-				if t.config.Debug {
-					evt, err := netPacketProtocolHandler(netDecoder, netEventMetadata, networkThread, "net_packet")
-					if err != nil {
-						t.handleError(err)
-						continue
-					}
-
-					// output the event
-					select {
-					case t.config.ChanEvents <- evt:
-						t.stats.NetEvCount.Increment()
-					case <-ctx.Done():
-						return
+				ifaceName := t.netInfo.ifaces[int(netCaptureData.InterfaceIndex)].Name
+				ifaceIdx, err := t.getTracedIfaceIdx(ifaceName)
+				if err == nil {
+					if t.eventsToTrace[netEventMetadata.NetEventId] && ifaceIdx >= 0 {
```

Comment:
I am a bit confused from the check of `ifaceIdx >= 0`.
Isn't it given if there is no error?